PocketCastsTests/Tests/Utilities/PodcastManagerTests.swift

```
@@ -0,0 +1,92 @@
+import XCTest
+import FMDB
+import PocketCastsDataModel
+@testable import podcasts
+
+final class PodcastManagerTests: XCTestCase {
+
+    private func setupDatabase() throws -> DataManager {
+        let dbQueue = try XCTUnwrap(FMDatabaseQueue.newTestDatabase())
+        return DataManager(dbQueue: dbQueue)
+    }
+
+    private var dataManager: DataManager!
+    private var downloadManager: DownloadManager!
+    private var podcast: Podcast!
+    private var episode: Episode!
+
+    override func setUp() async throws {
+        try await super.setUp()
+
+        let dataManager = try setupDatabase()
+        let downloadManager = DownloadManager(dataManager: dataManager)
+
+        let podcast = Podcast()
+        podcast.uuid = UUID().uuidString
+        podcast.subscribed = 0
+        podcast.addedDate = Date().addingTimeInterval(-1.week)
+        podcast.syncStatus = SyncStatus.synced.rawValue
+
+        dataManager.save(podcast: podcast)
+
+        let episode = Episode()
+        episode.uuid = UUID().uuidString
+        episode.podcastUuid = podcast.uuid
+        episode.podcast_id = podcast.id
+        episode.addedDate = podcast.addedDate
+        episode.downloadUrl = "http://google.com"
+        episode.playingStatus = PlayingStatus.notPlayed.rawValue
+
+        dataManager.save(episode: episode)
+        self.dataManager = dataManager
+        self.downloadManager = downloadManager
+        self.episode = episode
+        self.podcast = podcast
+    }
+
+    func testTaskCancellationForUnknownDeletion() async throws {
+        let podcastManager = PodcastManager(dataManager: dataManager, downloadManager: downloadManager)
+
+        // Verify the podcast and episode exist in the data manager after being added in `setUp`
+        XCTAssertEqual(dataManager.findPodcast(uuid: podcast.uuid, includeUnsubscribed: true), podcast)
+        XCTAssertEqual(dataManager.findEpisode(uuid: episode.uuid), episode)
```

Isn't that kinda of a test to make sure the Mock Data Manager works? Do we need to test that here? Wouldn't the rest of the asserts fail in case this doesn't work?

---

Yeah, it is, although this isn't the `DataManagerMock`, but a full separate test database instance.

I wondered if this is covering too much but when I was writing the test, any mismatch in the sample podcast or episode would cause the episode or podcast not to exist (failed database save, which is just logged to the console). In that case, the assertion to check for a running task would probably catch it but it didn't seem very clear.

I could also, or instead of these assertions, add a check that the first task is for the URL we expect, which would at least cover the case where some other tasks sneaks in.
```swift
XCTAssertEqual(task.originalRequest?.url?.absoluteString, episode.downloadUrl)
```